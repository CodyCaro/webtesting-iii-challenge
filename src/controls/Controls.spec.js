// Test away!
// Test away!
import React from "react";
import { render, shallow } from "react-testing-library";
import "@testing-library/jest-dom/extend-expect";
import Controls from "./Controls";

// test("Gate defaults to Unlocked and Open", () => {
//   const wrapper = render(<Controls />);
//   expect(wrapper.baseElement).toMatchSnapshot();
// });

test("Gate cannot be closed or opened if locked", () => {
  const wrapper = render(<Controls locked={true} />);
  //   const openCloseButton = wrapper.queryByText();
  //   console.log(wrapper);
  const openCloseButton = wrapper.queryByTestId("open-close");
  expect(openCloseButton).toHaveAttribute("disabled");
});
