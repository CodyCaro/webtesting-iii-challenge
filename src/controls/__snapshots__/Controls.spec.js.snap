// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gate defaults to Unlocked and Open 1`] = `
<body>
  <div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        data-testid="locked-unlocked"
        disabled=""
      >
        Unlock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="open-close"
        disabled=""
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
